src/Hangfire.Core/BackgroundJobServer.cs

```diff
@@ -163,5 +174,46 @@ public void Start()
         public void Stop()
         {
         }
+
+        public void OnNext(int notUsed)
+        {
+            var threshold = 0;
+            var delay = 0;
+
+            int.TryParse(ConfigurationManager.AppSettings["failedJobCountThreshold"], out threshold);
+            int.TryParse(ConfigurationManager.AppSettings["failedJobCheckInterval"], out delay);
+
+            TimeSpan duration = new TimeSpan(0, delay, 0);
+
+            lock (_failedLock)
+            {
+                _failedJobsCount++;
+            }            
+
+            if (!_hasFirstFailedDateTime || DateTime.Now.Subtract(duration) > _lastEmailNotification)
+            {
+                _lastEmailNotification = DateTime.Now;
+                _failedJobsCount = 1;
+                _hasFirstFailedDateTime = true;
+            }
+            else if(_failedJobsCount > threshold )
+            {
+                var name = Environment.MachineName;
+                var msg = $"The server with machine name: \"{ name }\" has had a peak of failed jobs.";
+
+                ServerStatusNotifier.Notify(1, msg);
+                _hasFirstFailedDateTime = false;
+            }
+        }
+
+        public void OnError(Exception error)
+        {
+            throw new Exception("Method \"OnError()\" not implemented.");
+        }
+
+        public void OnCompleted()
+        {
+            throw new Exception("Method \"OnCompleted()\" not implemented.");
```

Comment:
I don't mean put a text there I mean will this ever throw an exception because it will eventually complete and then break? logic doesn't seem right to me.